module/zfs/arc.c

```diff
@@ -9975,7 +9970,7 @@ l2arc_dev_hdr_read(l2arc_dev_t *dev)
 
 	err = zio_wait(zio_read_phys(NULL, dev->l2ad_vdev,
 	    VDEV_LABEL_START_SIZE, l2dhdr_asize, abd,
-	    ZIO_CHECKSUM_LABEL, NULL, NULL, ZIO_PRIORITY_ASYNC_READ,
+	    ZIO_CHECKSUM_LABEL, NULL, NULL, ZIO_PRIORITY_SYNC_READ,
```

Comment:
This makes sense, was the use of async here merely causing a reporting issue?

Reply:
No it wasn't causing any issue. This is a synchronous read (zio_wait) and I thought it would be more correct to have the corresponding flag, or am I wrong?

Reply:
No you're right.  I just thought you might have noticed it was being reported incorrectly in something like thew `zpool status -r` output which breaks down the IOs in part by priority.  Sounds like you just noticed it by inspection which is great.